jetstream_pt/third_party/llama/model_exportable.py

```diff
@@ -109,6 +118,21 @@ def __init__(
     )
     self.ffn_norm = RMSNorm(args.dim, eps=args.norm_eps, device=args.device)
 
+    self.hf_name("attention", "self_attn")
+    self.attention.hf_name("wq", "q_proj")
+    self.attention.hf_name("wk", "k_proj")
+    self.attention.hf_name("wv", "v_proj")
+    self.attention.hf_name("wo", "o_proj")
+
+    self.attention.annotate_sharding("wq.weight", 0)
+    self.attention.annotate_sharding("wk.weight", 0)
+    self.attention.annotate_sharding("wv.weight", 0)
```

Comment:
Since the weights are loaded as original model name, can we use yaml file sharding config? Additional sharding annoation add complexity.

Reply:
I am thinking of getting rid of that file.

Reason is because we want least number of commandline flags. So we need to somehow map model names to filename, then we still need to know the basedir, and handle cases when this file is not found.

Creation of this file actually depends on some information that is hardcoded in the model itself (the `get_weight_sharding_type` method). Reconciling these information together will make simpler code. 

I will remove all those when all models supports new cli and also remove `run_server` and `run_interactive` scripts. 

Right now they are kept so old run server / convert weights still work.

Reply:
Sounds good.